pkg/util/pod/pod.go

```diff
@@ -56,3 +63,36 @@ func gateIndex(p *corev1.Pod, gateName string) int {
 		return g.Name == gateName
 	})
 }
+
+func ReadUIntFromLabel(obj client.Object, labelKey string) (*int, error) {
+	return ReadUIntFromLabelWithMax(obj, labelKey, math.MaxInt)
+}
+
+func ReadUIntFromLabelWithMax(obj client.Object, labelKey string, max int) (*int, error) {
```

Comment:
I suggest to name to `ReadUIntFromLabelBelowBound` to make the inequality < bound. This way we don't need to remember to "-1" when calling the function.

Reply:
Done